package/MDAnalysis/coordinates/base.py

```diff
@@ -1282,6 +1305,156 @@ def __repr__(self):
                     nframes=self.n_frames,
                     natoms=self.n_atoms
                 ))
+                
+    def add_auxiliary(self, auxname=None, auxdata=None, **kwargs):
+        """Add auxiliary data to be read alongside trajectory.
+
+        Auxiliary data may be any data timeseries from the trajectory additional
+        to that read in by the trajectory reader. *auxdata* can be an 
+        :class:`~MDAnalysis.auxiliary.base.AuxReader` instance, or the data 
+        itself as e.g. an array, filename; in the latter case an appropriate 
+        :class:`~MDAnalysis.auxiliary.base.AuxReader` is guessed from the 
+        data/file format. An appropriate *format* may also be directly provided 
+        as a key word argument.
+
+        On adding, the AuxReader is initially matched to the current timestep
+        of the trajectory, and will be updated when the trajectory timestep
+        changes (through a call to :meth:`next()` or jumping timesteps with 
+        ``trajectory[i]``).
+
+        The representative value of the auxiliary data for each timestep (as
+        calculated by the :class:`~MDAnalysis.auxiliary.base.AuxReader`) is 
+        stored in current timestep in the ``ts.aux`` namespace under *auxname*; 
+        e.g. to add additional pull force data stored in pull-force.xvg::
+
+            u = MDAnalysis.Universe(PDB, XTC)
+            u.trajectory.add_auxiliary('pull', 'pull-force.xvg')
+
+        The representative value for the current timestep may then be accessed 
+        as ``u.trajectory.ts.aux.pull``.
+
+        See Also
+        --------
+        :meth:`remove_auxiliary`
+        :meth:`next_as_aux`
+        :meth:`iter_as_aux`
+
+        Note
+        ----
+        Auxiliary data is assumed to be time-ordered, with no duplicates. See 
+        the :ref:`Auxiliary API`.
+        """
+        # making auxname and auxdata keyworded so we can more easily reload
+        # an auxiliary; so we now have to check we have both
+        if not auxname or not auxdata:
```

Comment:
Just do not give a default value to these arguments and the interpretor will complain by itself if they are missing. It will not prevent to specify them as kwargs.